### PR TITLE
Support custom edit link dialog

### PR DESCRIPTION
**Pull request by hieutbui**
## Issue
- As we hardcode the editor's height, the default edit link dialog won't be fully displayed on small screens and editor will also can't scroll down. So we need to support a custom dialog displayed by the application using this library.
## Resolved

https://github.com/linagora/html-editor-enhanced/assets/80142234/300ba6fa-b2a2-4219-b157-fb5c3716aeaa

### CODE REVIEW

File: lib/src/widgets/html_editor_widget_web.dart [hunk 1]
```diff
@@ -818,6 +825,30 @@ class _HtmlEditorWidgetWebState extends State<HtmlEditorWidget> {
           });\n
         """;
     }
+    if (c.onEditLink != null) {
+      callbacks =
+      """$callbacks          \$('#summernote-2').on('summernote.mouseup', function(_, context) {
+            \$('.note-link-popover').off('click').on('click', function(e) {
+              if (e.target.className == 'note-icon-link' || (e.target.className == 'note-btn' && e.target.getAttribute('aria-label') == 'Edit')) {
```
Comment (dab246) [inline lib/src/widgets/html_editor_widget_web.dart:832]: I'm worried if the editor supports multiple languages. Is the value `e.target.getAttribute('aria-label')` always equal to `'Edit'`?

File: example/lib/main.dart [hunk 2]
```diff
@@ -162,6 +163,155 @@ class _HtmlEditorExampleState extends State<HtmlEditorExample> {
                   print('pasted into editor');
```
Comment (hoangdat) [inline example/lib/main.dart:163]: please remove it or keep it on debug only

File: example/lib/main.dart [hunk 3]
```diff
@@ -162,6 +163,155 @@ class _HtmlEditorExampleState extends State<HtmlEditorExample> {
...
                 }, onScroll: () {
                   print('editor scrolled');
```
Comment (hoangdat) [inline example/lib/main.dart:165]: idem

File: example/lib/main.dart (hunk 2 above)
Comment (hieutbui) [reply example/lib/main.dart]: done

File: example/lib/main.dart (hunk 3 above)
Comment (hieutbui) [reply example/lib/main.dart]: done

File: lib/src/widgets/html_editor_widget_web.dart (hunk 1 above)
Comment (hieutbui) [reply lib/src/widgets/html_editor_widget_web.dart]: I found better way is to check the `targetChildren` tag if `targetClass` is `note-btn`. It took care of the issue.

**Comment by dab246** [comment]
@hieutbui  Please rebase it

**Review by dab246 — APPROVED** [review]

**Pull request merged**